game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameSelectorPanel.java

```diff
@@ -312,30 +318,41 @@ public void update(final Observable o, final Object arg) {
   }
 
   private void selectSavedGameFile() {
-    GameFileSelector.selectGameFile(JOptionPane.getFrameForComponent(this))
+    GameFileSelector.builder()
+        .fileDoesNotExistAction(
+            file ->
+                DialogBuilder.builder()
+                    .parent(this)
+                    .title("Save Game File Not Found")
+                    .errorMessage("File does not exist: " + file.getAbsolutePath())
+                    .showDialog())
+        .build()
+        .selectGameFile(JOptionPane.getFrameForComponent(this))
         .ifPresent(
-            file -> {
-              try {
-                if (!GameRunner.newBackgroundTaskRunner()
-                    .runInBackgroundAndReturn(
-                        "Loading savegame...",
+            file ->
+                TaskRunner.builder()
+                    .waitDialogTitle("Loading Save Game")
+                    .exceptionHandler(
+                        e ->
+                            SwingComponents.showDialogWithLinks(
+                                DialogWithLinksParams.builder()
+                                    .title("Failed To Load Save Game")
+                                    .dialogType(DialogWithLinksTypes.ERROR)
+                                    .dialogText(
+                                        String.format(
+                                            "<html>Error: %s<br/><br/>"
+                                                + "If this is not expected, please "
+                                                + "file a <a href=%s>bug report</a><br/>"
+                                                + "and attach the error message above and the "
+                                                + "save game you are trying to load.",
```

Comment:
Indeed, though, swing is not really a compliant HTML parser and we're not really rendering HTML pages. It would be inappropriate to have a body or head tag. The prefix of `<html>` is effectively a hack to tell swing to do html rendering. IIRC, swing supports only a subset of tags.